pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/CancerStore.java

```diff
@@ -115,14 +117,24 @@ public CancerSuggestionResponse getCancerSuggestions(String cancerQuery, String
 
         // now rank the matches in a way that puts left-most matches near the top, favoring word start matches
         List<CancerSuggestion> sortedSuggestions = new ArrayList<>();
+        String regex = upperCancerQuery;
+        if (isSpanish) {
+            // search for the exact match or the match with Spanish accented chars substituted
+            regex = String.format("((%s)|(%s))", regex, asciiCharsReplacedWithSpanishChars);
+        }
+        Pattern pattern = Pattern.compile(regex);
         var suggestionComparator = new StringSuggestionTypeaheadComparator(upperCancerQuery);
         cancerMatches.stream()
                 .sorted((lhs, rhs) -> suggestionComparator.compare(lhs.getCancerName(), rhs.getCancerName()))
                 .limit(limit)
                 .forEach(cancer -> {
-                    int offset = cancer.getCancerName().toUpperCase().indexOf(upperCancerQuery);
-                    sortedSuggestions.add(new CancerSuggestion(new Cancer(cancer.getCancerName()),
-                            Collections.singletonList(new PatternMatch(offset, cancerQuery.length()))));
+                    Matcher matcher = pattern.matcher(cancer.getCancerName().toUpperCase());
+                    if (matcher.find()) {
```

Comment:
The start position and length should be calculated using the regex-based match, not the indexOf() and length of the query string.